connectors/src/api/slack_channels_linked_with_agent.ts

```diff
@@ -0,0 +1,157 @@
+import { Request, Response } from "express";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import { Op } from "sequelize";
+
+import { APIErrorWithStatusCode } from "@connectors/lib/error";
+import { SlackChannel } from "@connectors/lib/models";
+import { apiError, withLogging } from "@connectors/logger/withlogging";
+
+const PatchSlackChannelsLinkedWithAgentReqBodySchema = t.type({
+  agent_configuration_id: t.string,
+  slack_channel_ids: t.array(t.string),
+  connector_id: t.string,
+});
+
+type PatchSlackChannelsLinkedWithAgentReqBody = t.TypeOf<
+  typeof PatchSlackChannelsLinkedWithAgentReqBodySchema
+>;
+
+type PatchSlackChannelsLinkedWithAgentResBody =
+  | { success: true }
+  | APIErrorWithStatusCode;
+
+const _patchSlackChannelsLinkedWithAgentHandler = async (
+  req: Request<
+    Record<string, string>,
+    PatchSlackChannelsLinkedWithAgentResBody,
+    PatchSlackChannelsLinkedWithAgentReqBody
+  >,
+  res: Response<PatchSlackChannelsLinkedWithAgentResBody>
+) => {
+  const bodyValidation = PatchSlackChannelsLinkedWithAgentReqBodySchema.decode(
+    req.body
+  );
+
+  if (isLeft(bodyValidation)) {
+    const pathError = reporter.formatValidationErrors(bodyValidation.left);
+    return apiError(req, res, {
+      api_error: {
+        type: "invalid_request_error",
+        message: `Invalid request body: ${pathError}`,
+      },
+      status_code: 400,
+    });
+  }
+
+  const {
+    connector_id: connectorId,
+    agent_configuration_id: agentConfigurationId,
+    slack_channel_ids: slackChannelIds,
+  } = bodyValidation.right;
+
+  const slackChannels = await SlackChannel.findAll({
+    where: {
+      slackChannelId: slackChannelIds,
+      connectorId,
+    },
+  });
+
+  const foundSlackChannelIds = new Set(
+    slackChannels.map((c) => c.slackChannelId)
+  );
+
+  const missingSlackChannelIds = Array.from(
+    new Set(slackChannelIds.filter((id) => !foundSlackChannelIds.has(id)))
+  );
+
+  if (missingSlackChannelIds.length) {
+    return apiError(req, res, {
+      api_error: {
+        type: "not_found",
+        message: `Slack channel(s) not found: ${missingSlackChannelIds.join(
+          ", "
+        )}`,
+      },
+      status_code: 404,
+    });
+  }
+
+  await SlackChannel.update(
```

Comment:
We should do these two updates in a transaction. Either everything goes true or nothing, but no half state.